### PR TITLE
DOCSP-26238: create keys from variables

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-26238
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-26238-key-from-variable/fundamentals/typescript/#referencing-keys-that-incorporate-variables

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/fundamentals/typescript.txt [hunk 1]
```diff
@@ -122,6 +121,45 @@ The type error raised by the preceding code snippet is as follows:
    ...
    Type '"Sylvilagus"' is not assignable to type 'Condition<"Canis" | "Felis">'.
 
+Creating Keys from Variables
+````````````````````````````
+
+If you need to query a collection or perform another operation using a
```
Comment (mongoKart) [inline source/fundamentals/typescript.txt:127]: Issue: [Style Guide discourages "using" on its own for ambiguity reasons](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#u)

```suggestion
If you need to query a collection or perform another operation with a
```

File: source/fundamentals/typescript.txt [hunk 2]
```diff
@@ -122,6 +121,45 @@ The type error raised by the preceding code snippet is as follows:
...
+key created from specified variables, you must use an ``as const``
```
Comment (mongoKart) [inline source/fundamentals/typescript.txt:128]: Issue: Pretty sure I get this, but the wording is a little strange. When I think of a key being "created", I think of the operation that sets a value into a key that never existed (like your code sample below). Does that same thing happen when just performing a query, like you mention in the previous line? In other words, is the `as const` assertion required _any time_ you're referring to a key via a variable, or just when creating the key?

I think you can also drop the word "specified" before "variables; it's likely assumed that if the user is using a variable, they have control over it.

Suggestion: Specify the cases in which this is relevant

File: source/fundamentals/typescript.txt [hunk 3]
```diff
@@ -122,6 +121,45 @@ The type error raised by the preceding code snippet is as follows:
...
+     name: string;
+     mealtimes: Mealtime[];
+   }
+   
+   interface Mealtime{
+     time: string;
+     amount: number;
+   }
+
+The following code snippet performs a find and update operation on a
```
Comment (mongoKart) [inline source/fundamentals/typescript.txt:149]: [Compound modifier](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#hyphens-in-compound-modifiers)

```suggestion
The following code snippet performs a find-and-update operation on a
```

**Review by mongoKart — CHANGES_REQUESTED** [review]
Nice. One question about the scope of use

File: source/fundamentals/typescript.txt (hunk 2 above)
Comment (rustagir) [reply source/fundamentals/typescript.txt:128]: By creating a key I just mean selecting one. The key used in the example may or may not exist -- the document is not given for us to confirm. I mean creating a key reference -- putting together a key (basically a path to a nested (it technically could be top level too) field name). `as const` is requires any time you refer to a key that incorporates a variable as a field/nested field/index/part of a field name/whatever else. I updated the subsection header and example introduction to hopefully make this more clear.

**Review by mongoKart — APPROVED** [review]
Perfect. Awesome work

**Pull request merged**